sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -549,15 +549,18 @@ class HDFSFileCatalog(
 
   refresh()
 
+  private def isInvalidFile(f: FileStatus): Boolean =
+    f.getPath.getName.startsWith("_") || f.getLen == 0
```

Comment:
Where is the place that does not handle empty file properly? Does that also handle file with no records (a csv file has only header line)